plugins/modules/pacemaker_resource.py

```diff
@@ -0,0 +1,238 @@
+          operation_option:
+            - interval=20
+'''
+
+RETURN = '''
+cluster_resources:
+    description: The cluster resource output message.
+    type: str
+    sample: "Assumed agent name ocf:heartbeat:IPaddr2 (deduced from IPaddr2)"
+    returned: always
+'''
+
+from ansible_collections.community.general.plugins.module_utils.module_helper import StateModuleHelper
+from ansible_collections.community.general.plugins.module_utils.pacemaker import pacemaker_runner
+
+
+class PacemakerResource(StateModuleHelper):
+    change_params = ('name', )
+    diff_params = ('name', )
+    output_params = ('status')
+    module = dict(
+        argument_spec=dict(
+            state=dict(type='str', default='present', choices=[
+                'present', 'absent']),
+            name=dict(type='str', required=True),
+            resource_type=dict(type='dict', options=dict(
+                resource_name=dict(type='str'),
+                resource_standard=dict(type='str'),
+                resource_provider=dict(type='str'),
+            )),
+            resource_option=dict(type='list', elements='str', default=list()),
+            resource_operation=dict(type='list', elements='dict', default=list(), options=dict(
+                operation_action=dict(type='str'),
+                operation_option=dict(type='list', elements='str'),
+            )),
+            resource_meta=dict(type='list', elements='str'),
+            resource_argument=dict(type='dict', options=dict(
+                argument_action=dict(type='str', choices=[
+                                   'clone', 'master', 'group', 'promotable']),
+                argument_option=dict(type='list', elements='str'),
+            )),
+            disabled=dict(type='bool', default=False),
+            wait=dict(type='int', default=300),
+        ),
+        required_if=[('state', 'present', ['resource_type', 'resource_option'])],
+        supports_check_mode=True
+    )
+    use_old_vardict = False
+    default_state = "present"
+
+    def __init_module__(self):
+        self.runner = pacemaker_runner(self.module)
+        self.does_not = "Error: resource or tag id '{0}' not found".format(
+            self.vars.name)
+        self.error_messages = ["Error: '{0}' already exists".format(self.vars.name), "Error: Resource '{0} does not exist".format(self.vars.name)]
+        self.vars.set('previous_value', self._get())
+        self.vars.set('_value', self.vars.previous_value, output=False, change=True)
+        self.vars.set_meta('value', initial_value=self.vars.previous_value)
+
+    def _process_command_output(self, fail_on_err, ignore_err_msg=""):
+        def process(rc, out, err):
+          if fail_on_err and rc != 0 and err and ignore_err_msg not in err:
+              self.do_raise(
+                'pcs failed with error (rc={0}): {1}'.format(rc, err))
+          out = out.rstrip()
+          self.vars.value = None if out == "" else out
+          return self.vars.value
```

Comment:
indentation should be 4, not 2:
```suggestion
            if fail_on_err and rc != 0 and err and ignore_err_msg not in err:
                self.do_raise('pcs failed with error (rc={0}): {1}'.format(rc, err))
            out = out.rstrip()
            self.vars.value = None if out == "" else out
            return self.vars.value
```